src/main/java/com/sumologic/jenkins/jenkinssumologicplugin/sender/LogListener.java

```diff
@@ -0,0 +1,42 @@
+package com.sumologic.jenkins.jenkinssumologicplugin.sender;
+
+import com.sumologic.jenkins.jenkinssumologicplugin.PluginDescriptorImpl;
+import com.sumologic.jenkins.jenkinssumologicplugin.integration.SearchAction;
+import hudson.Extension;
+import hudson.console.ConsoleLogFilter;
+import hudson.model.AbstractBuild;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+/**
+ * This class intercepts console output stream for every jenkins build and decorates it with additional functionality.
+ * Depending on configuration it can add two functionalities:
+ * 1) Add consistent timestamps to all output lines.
+ * 2) Forward all output logs to Sumo Logic.
```

Comment:
In addition to what Jacek wrote, there are plugins/dependencies that could do the same for us but we either way had to intercept whole log stream so can do it with one pass and without requireing external dependencies on other plugins.